### PR TITLE
feat(gitlab) fetch trigger jobs

### LINKED ISSUE #6135: [Feature][Gitlab] Parsing CI/CD trigger jobs
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

I would like to have the possibility to fetch the trigger jobs from GitLab CI/CD pipelines
https://docs.gitlab.com/ee/api/jobs.html#list-pipeline-trigger-jobs

### Description

In my company, we have a setup where basically a main pipeline is triggering a child pipeline, which is responsible for the actual deployment (and from it I can have the actual environment where I'm deploying)
I know that GitLab doesn't offer an API to fetch those like we're fetching the pipelines, but the jobs need to be fetched for every single pipeline with the endpoint
`projects/{{ ProjectId }}/pipelines/{{ PipelineId }}/bridges`

I think it would be cool if we could have a setting for enable/disable this on the UI

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Marco-De-Stefani**
### Summary
Add the trigger jobs to the jobs already fetched from GitLab 

### Does this close any open issues?
Closes #6135

### Screenshots
data fetched from a real GitLab instance:
image

The processing of the data after the collection is tested with the `trigger_job_test.go`

### CODE REVIEW

**Comment by abeizn** [comment]
I will merge it after the issue https://github.com/apache/incubator-devlake/issues/5920

**Comment by abeizn** [comment]
LGTM, to be merged after v0.21-beta1 released

**Review by abeizn — APPROVED** [review]

File: backend/plugins/gitlab/tasks/trigger_job_collector.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func GetAllPipelinesIterator(taskCtx plugin.SubTaskContext, collectorWithState *helper.ApiCollectorStateManager) (*helper.DalCursorIterator, errors.Error) {
+	db := taskCtx.GetDal()
+	data := taskCtx.GetData().(*GitlabTaskData)
+	clauses := []dal.Clause{
+		dal.Select("gp.gitlab_id, gp.gitlab_id as iid"),
+		dal.From("_tool_gitlab_pipelines gp"),
+		dal.Where(
+			`gp.project_id = ? and gp.connection_id = ? and gp.gitlab_id not in (select json_extract(tj.input, '$.GitlabId') as gitlab_id from _raw_gitlab_api_trigger_job tj)`,
```
Comment (d4x1) [inline backend/plugins/gitlab/tasks/trigger_job_collector.go:78]: When running this SQL,  table `_raw_gitlab_api_trigger_job` doesn't exist.

**Comment by d4x1** [comment]
@Marco-De-Stefani  I encounter a problem when running the latest main branch and the log is 
```
time="2024-01-29 19:39:08" level=info msg=" [pipeline service] [pipeline #8] [task #59] finished step: 17 / 35"
time="2024-01-29 19:39:08" level=info msg=" [pipeline service] [pipeline #8] [task #59] executing subtask collectApiTriggerJobs"
time="2024-01-29 19:39:08" level=info msg="\x1b[31;1m/Users/houlinwei/Code/go/src/github.com/merico-dev/lake-feat-6854/backend/impls/dalgorm/dalgorm.go:188 \x1b[35;1mError 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist\n\x1b[0m\x1b[33m[2.624ms] \x1b[34;1m[rows:-]\x1b[0m SELECT gp.gitlab_id, gp.gitlab_id as iid FROM _tool_gitlab_pipelines gp WHERE gp.project_id = 50751699 and gp.connection_id = 1 and gp.gitlab_id not in (select json_extract(tj.input, '$.GitlabId') as gitlab_id from _raw_gitlab_api_trigger_job tj)"
time="2024-01-29 19:39:08" level=error msg=" [pipeline service] [pipeline #8] [task #59] subtask collectApiTriggerJobs ended unexpectedly\n\tWraps: (2) Error 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist (500)\n\tWraps: (3) Error 1146 (42S02): Table 'lake._raw_gitlab_api_trigger_job' doesn't exist\n\tError types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *mysql.MySQLError"
time="2024-01-29 19:39:08" level=info msg=" [task service] run task #61 in background "

```

And I think SQL in function `GetAllPipelinesIterator` is the root cause. Please check the codes. 
cc @abeizn

**Pull request merged**